File: java-project-euler-100/src/main/java/net/talaatharb/projecteuler100/P016.java
package net.talaatharb.projecteuler100;

import java.math.BigInteger;

/**
 * P016
 * 
 * @author mharb
 *
 */
public class P016 extends Solution {

	/**
	 * Problem solution entry point
	 * 
	 * @param args
	 */
	public static void main(String[] args) {
		new P016().run();
	}

	@Override
	public Number solve() {
		problemNumber = 16;
		int result = 0;
		final int n = 1000;
		final BigInteger power = BigInteger.valueOf(2).pow(n);
		final String powerString = power.toString();
		for (char i : powerString.toCharArray()) {
			result += Character.getNumericValue(i);
		}

		return result;
	}

}

File: typescript-project-euler-100/src/p016.ts
import { Solution } from "./Solution";

export class P016 extends Solution {

    solve() {
        this.problemNumber = 16;

		let result = 0;
		const n = 1000;
		const power = BigInt(Math.pow(2, n));
		const powerString = power.toString();
		for (const i of powerString.split('')) {
			result += Number(i);
		}

		return result;
    }
}
